### DISCUSSION TITLE
formally supported method to iterate through all commits

### DISCUSSION THREAD

**Discussion by fche** [gitpython-developers/GitPython#1425] [Ideas]
Repo.iter_items / iter_commits normally limits itself to parents of HEAD.
However, if a GitPython program wants to iterate through all commits in a git repo, it can use a hack like this:

       r = git.Repo('/path')
       for commit in r.iter_commits('--all'):
            pass

expecting the "--all" flag to be passed down to "git rev-list", and thence back to the python code.   It would be nice if there were a more natural interface to expose this search, one that could also work with GitDB.

**Comment by Byron** [comment]
Thanks for sharing your thoughts!

I turned this into a discussion in case there should be some brainstorming about how such API should look like with potential contributors.

Please note that `GitDB`, to me, is end of life, so the GitCmdDB would be the only implementation left to support.

This also means that passing `--all` is valid and working, even though it certainly isn't the most desirable/idiomatic way to achieve this.
